src/metabase/integrations/slack.clj

```diff
@@ -217,8 +224,11 @@
   "Calls Slack API `conversations.list` and returns list of available 'conversations' (channels and direct messages).
   By default only fetches channels, and returns them with their # prefix. Note the call to [[paged-list-request]] will
   only fetch the first [[max-list-results]] items."
-  [& {:as query-parameters}]
-  (let [params (merge {:exclude_archived true, :types "public_channel"} query-parameters)]
+  [& {:keys [oauth-scopes query-parameters]}]
+  (let [types  (if (contains? oauth-scopes "groups:read")
```

Comment:
nit: wdyt if we calling the oath-scopes here instead of getting it from the params? the `conversation-list` is only once and looks like we don't have a reason to exclude private channels if we have the scope.

Reply:
Hmm I did consider that.

The main problem I see is the tests, everything that fakes the `channel.list` endpoint will also need to fake the `auth.test` endpoint.

Curious to see what your take on this testing observation is?

One thing I thought about that might be better, rather than passing the scopes and have the function decide what that means is to pass a `:private-channels true` flag or fully externalize the new behaviour and have the caller override the `:types` parameter. I feel this would be a bit more in keeping with the previous version of the function. Not sure if that would feel more satisfying?

Reply:
yea, the testing is kinda suck if we have to mock the auth.test endpoint. I like the `:private-channels` flag idea, it's explicit and we can avoid mocking.